### DISCUSSION TITLE
1주차 사전과제, 신윤호

### DISCUSSION THREAD

**Discussion by syh1e** [MrMirror21/2024-KHU-CSE-TrackStudy-Web#15] [General]
1. 명명 규칙

- 변수나 함수 이름을 작성할 때, 첫 번째 단어는 소문자로 시작하고, 그 뒤의 각 단어의 첫 글자는 대문자로 씁니다.
- 모든 단어를 소문자로 쓰고 단어 사이를 밑줄(_)로 구분합니다.
- 단어 사이를 하이픈(-)으로 구분합니다.
- 변수 이름에 의미를 더하기 위해 특정 접두사나 접미사를 사용할 수 있습니다.
- 변수나 함수의 이름은 해당 요소가 무엇을 의미하는지 명확하게 드러내야 합니다. 변수가 나타내는 값이나 함수가 하는 일을 짐작할 수 있도록 이름을 짓는 것이 중요합니다.

2. 함수가 순수하다

부작용이 없고 같은 입력이 주어지면 항상 같은 출력을 반환하는 함수를 뜻합니다.

3. 순수 함수로만 모든 코드를 작성하는 것에 대한 생각

아직 개발을 안 해봐서 순수 함수로만 코드를 작성하는게 좋다고 생각합니다.

4. 함수 작성하는 방법

함수 표현식
```
const functionName = function(parameters) {
    // 함수 본문
};
```
함수의 이름이 필요하지 않은 경우
조건에 따라 함수를 정의하고 변수에 할당하는 경우
클로저를 생성하는 경우 

화살표 함수
```
const functionName = (parameters) => {
    // 함수 본문
};
```
함수가 단순한 경우 (중괄호와 return 생략 가능)
메소드 정의 시 사용
콜백 함수로 사용하는 경우

5. f-string

```
let x = 1
let y = 2
const pl = `${x}+${y}는 ${x+y}입니다.`
console.log(pl)
// 따옴표가 아니라 1옆에있는 기호 사용
```

6. 객체(object)에서 p 값 출력하기

주어진 코드와 같은 경우 객체 'o' 에서 'p'를 직접 'p'로 접근하는 방법은 없습니다.

**Comment by MrMirror21** [comment]
답변해주신 사전 과제 잘 봤습니다.
1번의 경우 명명 규칙에 관해서는 답변 잘해주셨는데, 윤호님이 이전에 어떤 명명규칙을 사용하셨는지도 적으시면서 생각해보시면 좋을 것 같습니다.
5번의 경우 잘 답변해주셨고, 1열에 있는 기호인 ` 은 백틱(backtick) 이라고 합니다.
6번의 경우 자바스크립트 문법 퀴즈였는데, 정답은 스터디 중에 언급 드릴테니 잘 들어주셨으면 합니다.
